rosetta/rosetta/projects/maxtext/README.md

```diff
@@ -0,0 +1,83 @@
+
+The [NVIDIA Container Toolkit](https://github.com/NVIDIA/nvidia-container-toolkit) is required to run the subsequent commands with GPU support. Ensure the NVIDIA Container Toolkit is installed before proceeding.
+
+## Containers
+We provide a fully built and ready-to-use multi-arch container which includes the latest optimizations, experimental features, and examples benchmarked for multi-node, multi-GPU training: `nvcr.io/nvidia/jax:24.04-maxtext-py3` (amd64 support). Verified containers will be updated periodically, but if you wish to use the bleeding edge (which may come with unexpected behavior), please use `ghcr.io/nvidia/jax:maxtext`. We also provide nightly dated images with the naming pattern `ghcr.io/nvidia/jax:maxtext-YYYY-MM-DD`, but we encourage you to use the latest ones for the best performance.
+
+*Note*: All paths mentioned in subsequent sections are relative to the top-level directory of the MaxText repository. When working interactively with containers, make sure you navigate to `/opt/maxtext` before running any commmands.
+
+## Downloading the C4 dataset
+You can use `download_dataset.sh` script to download the C4 dataset. For details regarding the dataset download please see these [guidelines](https://github.com/google/maxtext/blob/main/getting_started/First_run.md). Alternatively, you can pass set this argument `dataset_type=synthetic` while launching the training script to use a synthetic dataset which is quite helpful to debug the initial performance. For more datasets, please see the [Paxml readme](https://github.com/NVIDIA/JAX-Toolbox/blob/main/rosetta/rosetta/projects/pax/README.md#downloading-the-pile-and-lambada-datasets).
+
+## Launching a container
+Use the following command to launch a container:
+```
+docker run -ti --gpus=all --net=host --ipc=host -v <WORKSPACE_PATH>:/opt/maxtext/workspace -w /opt/maxtext <CONTAINER> /bin/bash
+```
+where `WORKSPACE_PATH` is the path to the directory where you would like to store any persistent files and `container` is the name of the maxtext container. You can additionally add dataset and vocab paths with the `-v` flag.
+
+## Running a job
+### Quick Runs
+#### Interactive: Single node
+Once the container is up and running, you can quickly launch a job with the following command
+```
+python3 MaxText/train.py MaxText/configs/base.yml hardware=gpu run_name=$YOUR_JOB_NAME
+```
+You can similarly launch a llama2-7b training job with following command:
+```
+python3 MaxText/train.py MaxText/configs/base.yml run_name=$YOUR_JOB_NAME use_iota_embed=true scan_layers=true\
+    steps=15 per_device_batch_size=2 model_name=llama2-7b remat_policy=minimal_flash enable_checkpointing=false logits_dot_in_fp32=false\
+    base_output_directory=local_train dataset_path=local dataset_type=synthetic attention=dot_product\
+    max_target_length=4096 hardware=gpu
```

Comment:
nit: Could you change it to one arg per line? It makes reading/updating a little easier.
```suggestion
python3 MaxText/train.py \
    MaxText/configs/base.yml \
    run_name=$YOUR_JOB_NAME \
    use_iota_embed=true scan_layers=true \
    steps=15 \
    per_device_batch_size=2 \
    model_name=llama2-7b \
    remat_policy=minimal_flash enable_checkpointing=false \
    logits_dot_in_fp32=false \
    base_output_directory=local_train \
    dataset_path=local \
    dataset_type=synthetic attention=dot_product \
    max_target_length=4096 \
    hardware=gpu
```

Reply:
Also, does this use multiprocess by default?

Reply:
No, the interactive run does not use multiprocess